oxidation/client/src/views/SettingsPage.vue

```diff
@@ -99,34 +99,35 @@ import {
 } from '@ionic/vue';
 
 import SettingsOption from '@/components/SettingsOption.vue';
-import { ref } from 'vue';
+import { ref, inject, toRaw } from 'vue';
 import { useI18n } from 'vue-i18n';
-import { Storage } from '@ionic/storage';
 import { onMounted } from '@vue/runtime-core';
 import { toggleDarkMode } from '@/states/darkMode';
+import { Config, StorageManager } from '@/services/storageManager';
 
 const { locale } = useI18n();
-const theme = ref('system');
-const enableTelemetry = ref(false);
-const minimizeToSystemTray = ref(false);
-const store = new Storage();
+const storageManager: StorageManager = inject('storageManager')!;
+const config = ref<Config>(structuredClone(StorageManager.DEFAULT_CONFIG));
 
 async function changeLang(selectedLang: string): Promise<void> {
+  config.value.locale = selectedLang;
   locale.value = selectedLang;
-  await store.create();
-  await store.set('userLocale', selectedLang);
+  await storageManager.storeConfig(toRaw(config.value));
 }
 
 async function changeTheme(selectedTheme: string): Promise<void> {
-  theme.value = selectedTheme;
+  config.value.theme = selectedTheme;
   toggleDarkMode(selectedTheme);
-  await store.create();
-  await store.set('userTheme', selectedTheme);
+  await storageManager.storeConfig(toRaw(config.value));
 }
 
 onMounted(async (): Promise<void> => {
-  await store.create();
-  theme.value = await store.get('userTheme') || 'system';
+  config.value = await storageManager.retrieveConfig();
+
+  if (!config.value.theme) {
+    config.value.theme = 'system';
+  }
+  locale.value = config.value.locale;
```

Comment:
Why this line was added? Is this the cause of the following problem:
![image](https://user-images.githubusercontent.com/19772607/219051912-68a15425-b00b-467a-add4-118aab93eb17.png)

Reply:
Mistake.

Reply:
LGTM